crates/router/src/core/pm_auth.rs

```diff
@@ -357,7 +357,24 @@ async fn store_bank_details_in_payment_methods(
     let mut new_entries: Vec<storage::PaymentMethodNew> = Vec::new();
 
     for creds in bank_account_details_resp.credentials {
-        let hash_string = format!("{}-{}", creds.account_number, creds.routing_number);
+        let (account_number, hash_string) = match creds.account_details {
+            pm_auth_types::PaymentMethodTypeDetails::Ach(ach) => (
+                ach.account_number.clone(),
+                format!(
+                    "{}-{}",
```

Comment:
Lets include additional metadata like the payment method type (Bacs, Sepa), in the hash.